### PR TITLE
Fixing the "Broken link in documentation" (Closes #653, again)

### LINKED ISSUE #653: Broken link in documentation
Here: https://github.com/corona-warn-app/cwa-app-android/blob/master/docs/architecture-overview.md

Link to https://github.com/corona-warn-app/cwa-server/blob/master/docs/architecture-overview.md does not work, page not found.

### PR DESCRIPTION
**Pull request by pr1metine**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [x] Make sure that your PR does not contain changes in text files, therefore the PR must not contain changes in values/strings/* and / or assets/* (see issue #332 for further information)
* [x] Make sure that your PR does not contain compiled sources (already set by the default .gitignore) and / or binary files

## Description

Basically #655. This time I used my free plan of Grammarly and corrected many punctuation mistakes. Closes #653 for good.

### CODE REVIEW

**Comment by pr1metine** [comment]
I really do not know what I am doing wrong. For now, I contacted CircleCI support to look into it.

Maybe, you guys could make a commit into this fork.

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,5 +48,4 @@ workflows:
   version: 2
   workflow:
     jobs:
-      - quickBuildReleaseWithTestsAndChecks:
-          context: SonarCloud
+      - quickBuildReleaseWithTestsAndChecks
```
[the remark this answers is not in the document]
Comment (pr1metine) [reply .circleci/config.yml:51]: My bad, I just wanted to test some suggestions. I'm going to open a new PR regarding CI once I'm back from school.

[the remark this answers is not in the document]
Comment (pr1metine) [reply .circleci/config.yml:51]: Could you, @jakobmoellersap, or your colleges revert my changes on `.circleci/config.yml`? If I tried to revert them myself I would be more than likely to fail the CI checks.

**Comment by tkowark** [comment]
@pr1metine how about this proposal: You provide the PR with the fix for the Unauthorized error, we merge that, and then you update this branch?

**Comment by pr1metine** [comment]
Sounds good to me! I'll get to work in a few hours.

**Comment by pr1metine** [comment]
In case you guys don't like my changes, feel free to `cherry-pick` the best parts. At least pick [this commit](https://github.com/corona-warn-app/cwa-app-android/pull/677/commits/828c59fde8bf64508ecdab20fed8458a203f0a9d) though.

**Comment by pr1metine** [comment]
What's the hold-up? 

- Am I not supposed to make changes to text files? 
- Is there an internal discussion about this? 
- Should I also push these changes to the iOS repo?

**Comment by tkowark** [comment]
As suggested by you, we are cheery picking the most important commit in a separate PR
https://github.com/corona-warn-app/cwa-app-android/pull/890

Thanks again for your contribution!
